src/extensions/command/commandext.cpp

```diff
@@ -713,6 +715,237 @@ bool ToggleSuperTimersCommandClass::Process()
 }
 
 
+/**
+ *  Switches the sidebar to the Building Tab.
+ *
+ *  @author: ZivDero
+ */
+const char* SetStructureTabCommandClass::Get_Name() const
+{
+    return "StructureTab";
+}
+
+const char* SetStructureTabCommandClass::Get_UI_Name() const
+{
+    return "Select Building Tab";
+}
+
+const char* SetStructureTabCommandClass::Get_Category() const
+{
+    return Text_String(TXT_INTERFACE);
+}
+
+const char* SetStructureTabCommandClass::Get_Description() const
+{
+    return "Switch the command bar to the Building Tab and select the completed building if any.";
+}
+
+bool SetStructureTabCommandClass::Process()
+{
+    const SidebarClassExtension::SidebarTabType newtab = SidebarClassExtension::SIDEBAR_TAB_STRUCTURE;
+    bool result = SidebarExtension->Change_Tab(newtab);
+
+    /**
+     *  Enter the manual placement mode when a building is complete
+     *  and pending placement on the sidebar.
+     *
+     *  @author: CCHyper (based on research by dkeeton)
+     */
+    if (PlayerPtr)
+    {
+        /**
+         *  Fetch the houses factory associated with producing buildings.
+         */
+        FactoryClass* factory = PlayerPtr->Fetch_Factory(RTTI_BUILDING);
+        if (!factory)
+            return result;
+
+        /**
+         *  If this object is still being built, then bail.
+         */
+        if (!factory->Has_Completed()) {
+            return result;
+        }
+
+        TechnoClass* pending = factory->Get_Object();
+
+        /**
+         *  If by some rare chance the product is not a building, then bail.
+         */
+        if (pending->What_Am_I() != RTTI_BUILDING)
+            return result;
+
+        BuildingClass* pending_bptr = reinterpret_cast<BuildingClass*>(pending);
+
+        /**
+         *  Fetch the factory building that can build this object.
+         */
+        BuildingClass* builder = pending_bptr->Who_Can_Build_Me();
+        if (!builder)
+            return result;
+
+        /**
+         *  Are we already trying to place this building? No need to re-enter placement mode...
+         */
+        if (Map.PendingObjectPtr == pending_bptr)
+            return result;
```

Comment:
IMO makes sense to swap these and compare the pointer to pending object before calling `Who_Can_Build_Me`.

```suggestion
        /**
         *  Are we already trying to place this building? No need to re-enter placement mode...
         */
        if (Map.PendingObjectPtr == pending_bptr)
            return result;

        /**
         *  Fetch the factory building that can build this object.
         */
        BuildingClass* builder = pending_bptr->Who_Can_Build_Me();
        if (!builder)
            return result;
```